rtic-common/src/wait_queue.rs

```diff
@@ -141,6 +141,9 @@ impl<T: Clone> Link<T> {
     /// Remove this link from a linked list.
     pub fn remove_from_list(&self, list: &DoublyLinkedList<T>) {
         cs::with(|_| {
+            if self.is_popped() {
```

Comment:
Sorry to double-ask, will need @korken89 confirmation here: I think it'll also need to be after the compiler fence, to ensure that the atomic write is observed.

This is probably not a problem on ARM in practice, but for good measure :)